src/index.ts

```diff
@@ -204,7 +228,22 @@ export function lilconfig(
             };
 
             const searchItems = getSearchItems(searchPlaces, searchPaths);
-            for (const {searchPlace, filepath, loaderKey} of searchItems) {
+            const visited: Set<string> = new Set();
+            for (const {
+                searchPlace,
+                filepath,
+                loaderKey,
+                searchPath,
+            } of searchItems) {
+                if (cache) {
+                    if (searchCache.has(searchPath)) {
```

Comment:
If lilconfig is configured with multiple `searchPlaces`, this code will lookup for the same cache entry for each search place. In the case of Prettier, [14 different searchPlaces](https://github.com/prettier/prettier/blob/83b9f62db4111f7d6c3a7ebd2fc01fca237d9072/src/config/get-prettier-config-explorer.js#L11-L25) are passed. 

Since `Map.prototype.has` is very efficient, it shouldn't make a big difference performance-wise. But it be ideal to avoid this.

Reply:
I am using `searchPath` as a cache key, not `filepath`. On each iteration `searchPath` is the directory where we expect the config to be, not the full path to the config file. This keeps the code simpler and we will have less entries in the cache as a result. Example values from a loop iteration:

`searchPath` -> `/foo/bar/baz`
`filepath` -> `/foo/bar/baz/prettier.config.js`

I agree that the naming is rather confusing. I haven't came up with a better names.

Reply:
Thanks for the clarification. I think my comment is still valid, here is a concrete example:

```js
const resolver = lilconfig('config', {
  searchPlaces: ['package.json', '.prettierrc', '.prettierrc.json'],
});

resolver.search('./foo');
```

In the above example, `getSearchItems` would generate the following entries:
* `{ searchPath: './foo', filepath: './foo/package.json' }`
* `{ searchPath: './foo', filepath: './foo/.prettierrc' }`
* `{ searchPath: './foo', filepath: './foo/.prettierrc.json' }`
* `{ searchPath: './', filepath: './package.json' }`
* `{ searchPath: './', filepath: './.prettierrc' }`
* `{ searchPath: './', filepath: './.prettierrc.json' }`

Correct me if I am wrong but with the for loop iterating over the entries generated by `getSearchItems`, when cache is enabled, it appears to me that this code will do:
* 3 consecutive cache lookup with: `searchCache.get('./foo')`
* follow by, 3 consecutive cache lookup with: `searchCache.get('./')`

IMO, it's not a big deal performance-wise since the `has` check on a `Map` is not very expensive. But ideally, the code should do 1 lookup for each `searchpath` in the search cache.

Reply:
Thanks for expanding on this. I've switched the look up for two nested loops. No more unnecessary look ups